### ISSUE TITLE
Missing release notes from GitHub releases for my project

### ISSUE THREAD

**Issue by szpak** [dependabot/feedback#663]
For my [project](https://github.com/Codearte/gradle-nexus-staging-plugin) I [don't see](https://github.com/Codearte/gradle-nexus-staging-plugin/pull/129) release notes and commits in the PR created by Dependabot. As I use [GitHub releases](https://github.com/Codearte/gradle-nexus-staging-plugin/releases) (and in addition a synchronized [CHANGELOG.md](https://github.com/Codearte/gradle-nexus-staging-plugin/blob/master/CHANGELOG.md) file) I would be happy to get know what prevents Dependabot from getting all required data?

**Comment by greysteil** [comment]
Thanks for the feedback.

Looking at that PR, I can see it doesn't include a link to the source code for `gradle-nexus-staging-plugin`, either. That means Dependabot was unable to find that source code, which means it didn't have a chance of finding the release notes or changelog.

The logic Dependabot uses to try to find the source code for a Gradle dependency is [here](https://github.com/dependabot/dependabot-core/blob/master/gradle/lib/dependabot/gradle/metadata_finder.rb#L17-L29), and in particular [this method](https://github.com/dependabot/dependabot-core/blob/master/gradle/lib/dependabot/gradle/metadata_finder.rb#L52-L64) is relevant. Could you add a `project > url` or `project > scm > url`  node to the POM that is generated for your project? Or is there any other way you can think of that Dependabot ought to have been able to find its source code?

**Comment by szpak** [comment]
Thanks for looking into that so quickly!

I checked the `pom.xml` file and I have both `project > url` and `project > scm > url` nodes:
https://repo1.maven.org/maven2/io/codearte/gradle/nexus/gradle-nexus-staging-plugin/0.21.1/gradle-nexus-staging-plugin-0.21.1.pom
Could be a reason that the second one points to `https://github.com/Codearte/gradle-nexus-staging-plugin` instead of `https://github.com/Codearte/gradle-nexus-staging-plugin.git`?

**Comment by greysteil** [comment]
Hmmm, it looks like Dependabot is fetching the POM from https://plugins.gradle.org/m2/, which returns a slimmed down version of the POM without those details. What's the story there? What's the correct behaviour?

**Comment by szpak** [comment]
That's a very good observation.

Long explanation. If plugin is applied by the plugins block:
```
plugins {
  id "io.codearte.nexus-staging" version "0.21.1"
}
```
then `https://plugins.gradle.org/m2/` is used by default.

If the plugin is applied with the old syntax (sometimes it is required):
```
buildscript {
  repositories {
    maven {
      url "repoUrl"
    }
  }
  dependencies {
    classpath "io.codearte.gradle.nexus:gradle-nexus-staging-plugin:0.21.1"
  }
}
```
it dependes URL to which repository is defined - Plugins or Maven Central. Therefor, the correct behavior would be to detect that repository and use it. However:
1. Almost all Gradle plugins are in the Plugins repository
2. Currently Gradle can [reuse](https://github.com/gradle/gradle/issues/5930) `pom.xml` generated by the `maven-publish` plugin, so the same file should be in both places. In my case it doesn't work as I use an old `maven` plugin for publishing artifacts. Once migrated, it should be ok.

In conclusion, I think it's a minor case with workarounds and you have more important things to follow :). Closing.
